vcs/gitcmd/repo.go

```diff
@@ -343,6 +343,10 @@ func (r *Repository) commitLog(opt vcs.CommitsOptions) ([]*vcs.Commit, uint, err
 		args = append(args, "--skip="+strconv.FormatUint(uint64(opt.Skip), 10))
 	}
 
+	if opt.Path != "" {
+		args = append(args, "--follow")
+	}
+
 	// Range
 	rng := string(opt.Head)
```

Comment:
Just to be safe, I think it'd be good to check that opt.Head doesn't begin with `-` (in case git options change and this opens an attack vector). Callers might already be doing this, in which case document that callers MUST do this and then no need to do it here.